core/sync/futex_haiku.odin

```diff
@@ -0,0 +1,127 @@
+//+private
+package sync
+
+import "core:c"
+import "core:c/libc"
```

Comment:
Do not use libc here the atomics. package sync has all of the atomics stuff in it already.